src/util.nim

```diff
@@ -434,3 +442,82 @@ template chalkCloseStream*(chalk: ChalkObj) =
   chalk.streamRefCt = 0
 
   delByValue(cachedChalkStreams, chalk)
+
+type Redacted* = ref object
+  raw:      string
+  redacted: string
+
+proc redact*(raw: string): Redacted =
+  return Redacted(raw: raw, redacted: raw)
+
+proc redact*(raw: string, redacted: string): Redacted =
+  return Redacted(raw: raw, redacted: redacted)
+
+proc redact*(data: seq[string]): seq[Redacted] =
+  result = @[]
+  for i in data:
+    result.add(redact(i))
+
+proc redacted*(data: seq[Redacted]): seq[string] =
+  result = @[]
+  for i in data:
+    result.add(i.redacted)
+
+proc raw*(data: seq[Redacted]): seq[string] =
+  result = @[]
+  for i in data:
+    result.add(i.raw)
+
+proc replaceItemWith*(data: seq[string], match: string, sub: string): seq[string] =
+  result = @[]
+  for i in data:
+    if i == match:
+      result.add(sub)
+    else:
+      result.add(i)
+
+template get[T](data: seq[T], i: int, default: T): T =
+  if len(data) > i:
+    data[i]
+  else:
+    default
+
+iterator zipLongest*[T](data1, data2: seq[T], default: T): (T, T) =
```

Comment:
Personally, I think these semi-functional APIs value conciseness over clarity. You'll notice I never use anything like map()/filter()/reduce(). It's fine to stay in, but please try to avoid, or comment if you really think you need it.

Reply:
Yeah per above, you don't even need this, since you'd be better off keeping 3-tuples anyway.

Reply:
removed